yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtilSuite.scala

```diff
@@ -233,4 +233,17 @@ class YarnSparkHadoopUtilSuite extends SparkFunSuite with Matchers with Logging
       }
     assert(caught.getMessage === "Can't get Master Kerberos principal for use as renewer")
   }
+
+  test("check different hadoop utils based on env variable") {
+    import org.apache.spark.deploy.SparkHadoopUtil
```

Comment:
You can put this import up top I suppose, and `java.lang.*` classes never need to be prefixed. 

Rather than cast incorrectly on purpose and check an exception, just look at the `getSimpleName` of the class and compare to "YarnSparkHadoopUtil" ?